### PR TITLE
Feature/deploy vpc

### PR DESCRIPTION
**Pull request by Leo-Messen**
- TF Code added to base layer vpc.tf
- VPC deployed with Terraform Module 
- VPC configured with CIDR Range 10.0.0.0/16
- Public & Private Subnets in all available AZs
- All subnets configured with equal number of IPs with maximum number per subnet
- NAT Gateway enabled according to requirements
- VPC Deployed to eu-west-2 region

### CODE REVIEW

File: terraform/layers/base/vpc.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+locals {
+  az_names = slice(data.aws_availability_zones.available_azs.names, 0, 3)
+}
+
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+
+  name = var.lab-1-vpc-name
```
Comment (nben55) [inline terraform/layers/base/vpc.tf]: This approach is fine but might be worth naming resources with project and environment. E.g. 
`name = "${var.project_name}-${var.environment}-vpc"` 

or creating a local, an example would be 
```
locals {
  resource_name_prefix = "${var.project_name}-${var.environment}"
}
```

and naming resources like `name = "${local.resource_name_prefix}-vpc"`

Might help down the line - common approaches to help quickly see what reosources belong to which project / environment

**Review by nben55 — APPROVED** [review]
Looks good, added one nitpick comment on naming

File: terraform/layers/base/vpc.tf (hunk 1 above)
Comment (Leo-Messen) [reply terraform/layers/base/vpc.tf]: Makes sense, thank you!

**Review by nben55 — APPROVED** [review]

**Review by PAYNEA03 — APPROVED** [review]
looks good

**Pull request merged**